libyul/AsmAnalysis.cpp

```diff
@@ -766,3 +789,42 @@ bool AsmAnalyzer::validateInstructions(FunctionCall const& _functionCall)
 {
 	return validateInstructions(_functionCall.functionName.name.str(), nativeLocationOf(_functionCall.functionName));
 }
+
+bool AsmAnalyzer::validateObjectStructure(langutil::SourceLocation _astRootLocation)
+{
+	if (m_eofVersion.has_value())
+	{
+		if (util::contains(m_objectName, '.')) // No dots in object name for EOF
+		{
+			m_errorReporter.syntaxError(
+				9822_error,
+				_astRootLocation,
+				fmt::format(
+					"In EOF context object name \"{objectName}\" must not contain 'dot' character.",
+					fmt::arg("objectName", m_objectName)
+				)
+			);
+			return true;
+		}
+	}
+
+	return false;
+}
+
+std::set<std::string> AsmAnalyzer::getTopLeveleSubobjectNames(std::set<std::string> const& _names, std::string const& _objectName)
+{
+	std::set<std::string> topLevelObjectNames;
+
+	for (auto const& name: _names)
+	{
+		auto const objectNamePlusDotPos = name.find(_objectName + '.');
+		if (objectNamePlusDotPos == 0)
+		{
+			const auto remainigString = name.substr(objectNamePlusDotPos);
```

Comment:
The typo here is failing the `chk_coding_style` job.

Reply:
This could actually use better name and I'd also assert that all the absolute paths start with `_objectName`.

```c++
if (util::contains(subobjectPath, '.'))
{
	yulAssert(boost::starts_with(subobjectPath, _objectName + '.'));
	std::string const relativeSubobjectPath = subobjectPath.substr(_objectName.size() + 1);
	if (!util::contains(relativeSubobjectPath, '.'))
		topLevelObjectNames.insert(relativeSubobjectPath);
}
```

But I see that this is actually different from the version in #15536. CI did not fail there. You should move it there.